### PR TITLE
[MRG] exposed oversampling parameter

### PR DESCRIPTION
**Pull request by bthirion**
Addresses #182 
Please review

### CODE REVIEW

**Comment by bthirion** [comment]
Any review on that one ?

**Comment by KamalakerDadi** [comment]
Issue #182 hints out at the example "An example showcasing this behavior would actually be useful."

Any plans for an example.

**Comment by KamalakerDadi** [comment]
Can you add changes into doc/whats_new.rst ?

File: nistats/design_matrix.py [hunk 1]
```diff
@@ -274,7 +281,7 @@ def _full_rank(X, cmax=1e15):
 def make_design_matrix(
     frame_times, paradigm=None, hrf_model='glover',
     drift_model='cosine', period_cut=128, drift_order=1, fir_delays=[0],
-    add_regs=None, add_reg_names=None, min_onset=-24):
+    add_regs=None, add_reg_names=None, min_onset=-24, oversampling=None):
```
Comment (KamalakerDadi) [inline nistats/design_matrix.py]: In some functions oversampling is by default 16. Is there any reason to keep default to None and then convert to 16.

File: nistats/design_matrix.py [hunk 2]
```diff
@@ -332,6 +339,10 @@ def make_design_matrix(
         Minimal onset relative to frame_times[0] (in seconds)
         events that start before frame_times[0] + min_onset are not considered.
 
+    oversampling: float or None, optional,
+        Oversampling factor used in temporal convolutions. Should be 1 for
+        whenever hrf_mode is 'fir' and 16 otherwise.
```
Comment (KamalakerDadi) [inline nistats/design_matrix.py]: Typo here should be hrf_model

File: nistats/design_matrix.py [hunk 3]
```diff
@@ -191,6 +191,10 @@ def _convolve_regressors(paradigm, hrf_model, frame_times, fir_delays=[0],
         Minimal onset relative to frame_times[0] (in seconds) events
         that start before frame_times[0] + min_onset are not considered.
 
+    oversampling: float or None, optional,
+        Oversampling factor used in temporal convolutions. Should be 1 for
+        whenever hrf_mode is 'fir' and 16 otherwise.
```
Comment (KamalakerDadi) [inline nistats/design_matrix.py]: Same here typo hrf_model

File: nistats/design_matrix.py [hunk 4]
```diff
@@ -210,8 +214,11 @@ def _convolve_regressors(paradigm, hrf_model, frame_times, fir_delays=[0],
     regressor_names = []
     regressor_matrix = None
     if hrf_model == 'fir':
+        if oversampling not in [1, None]:
+            warn('Forcing oversampling factor to 1 for a finite'
+                 'impulse response hrf model')
```
Comment (KamalakerDadi) [inline nistats/design_matrix.py]: Can you increase the stacklevel here ? So that it won't float around other verbose messages.

**Comment by KamalakerDadi** [comment]
I don't have other comments as far as I can see.

**Comment by adelavega** [comment]
Looks good to me. Thanks for the PR

**Comment by adelavega** [comment]
Question: Is there a reason for the default parameter being 16? As I pointed out in my original issue, this is too low for high-frequency effects, and leads to odd behavior. Is there a downside to making the default a higher value? Or would this have unintended consequences for lower frequency events?

**Comment by bthirion** [comment]
I think I picked 16 because this is the SPM default. Making it larger would result in additional computational cost, but it's negligible. The true point is: in 99% of experimental designs people don't use very high frequency signals, as these fast changes will be killed by hrf convolution anyhow...
What deafult value do you advocate ? 32 ?

**Comment by adelavega** [comment]
We've been using 50, and that seems to work for natural speech (word speech rate). Personally, I'm happy to manipulate this parameter as I need, but I think a default that covers a larger set of use cases is always good (given negligible increased computation, relative to model fitting)

**Comment by bthirion** [comment]
"An example showcasing this behavior would actually be useful."
Yes, I wrote that. I'm currently refactoring the doc, and I think that exposing this parameter  is not a high priority, and I propose to drop it in the short term.

**Comment by bthirion** [comment]
More comments on that one ?

**Pull request merged**